### PR TITLE
Change Planet Suitiability alignment to spaces, add logging

### LINKED ISSUE #1936: Planet Suitability texts are not aligned
### Environment

* **FreeOrion Version**:
v0.4.7+[build 2018-01-02.8092012] MSVC 2015
* **Operating System**:
Windows 7 Pro
* **Graphic card used**: 
NVIDIA GeForce 940MX
* **Fetched as**
  * Weekly development build

### Description
In recent dev releases, the Planet Suitability texts are not aligned, looks messy.
planet suitability texts not aligned

### Expected Result
The suitability texts of the races shall be aligned.

### Steps to reproduce
* First step.
Right click any planet, open the "Planet Suitability" Pedia dialog.
* Second step.
The suitablility texts of the races are not aligned.

### PR DESCRIPTION
**Pull request by dbenage-cx**
Attempt to address #1936 with some added logging for future troubleshooting.

(Edit to prevent auto-close of issue)

### CODE REVIEW

File: UI/EncyclopediaDetailPanel.cpp [hunk 1]
```diff
@@ -2689,30 +2712,38 @@ namespace {
 
             if (it->first > 0) {
                 if (!positive_header_placed) {
-                    detailed_description += str(FlexibleFormat(UserString("ENC_SUITABILITY_REPORT_POSITIVE_HEADER"))
+                    auto pos_header = str(FlexibleFormat(UserString("ENC_SUITABILITY_REPORT_POSITIVE_HEADER"))
                                                 % planet->PublicName(planet_id));
```
Comment (geoffthemedio) [inline UI/EncyclopediaDetailPanel.cpp]: fix indenting

Comment (dbenage-cx) [reply UI/EncyclopediaDetailPanel.cpp]: Fixed

**Comment by dbenage-cx** [comment]
~The one remaining tab is in the stringtable entry `ENC_SPECIES_PLANET_TYPE_SUITABILITY_COLUMN1`~

~At least on my system, this tab is needed to align the text to a width divisible by the width of a space character. (e.g. a width of 10 vs a space width of 4)
Otherwise some text ends up with a slight offset less than the width of a space.
I left this in stringtables for easier troubleshooting if it is an issue on other platforms.~
(Edit: Tab now removed, leaving comment)

**Comment by geoffthemedio** [comment]
Interacts weirdly with drag-selecting text, but that's probably not an issue specific these changes...
drag_select_layout_weirdness

**Comment by dbenage-cx** [comment]
Just verified that behavior is consistent with release-v0.4.8 branch.

Removing the above mentioned tab prevents such behavior, marking as WIP to try some alternatives.

**Comment by Oberlus** [comment]
> Interacts weirdly with drag-selecting text, but that's probably not an issue specific these changes...

With tabs it also interacts weirdly with drag-selection.

**Comment by dbenage-cx** [comment]
Workaround to selection done with use of hair space (0x200A) to buffer the shorter entry width.
This was the only whitespace character I found to resolve with an extent width less than 4.

User selection of text containing a tab is still potentially an issue in other entries/places.

**Comment by dbenage-cx** [comment]
This PR is currently crashing on Win10 display of report with Invalid UTF8, but does not crash 100% of time

**Comment by dbenage-cx** [comment]
Rebased

**Comment by Vezzra** [comment]
@dbenage-cx, do you still get those Invalid-UTF8 crashes on Windows, or does the removal of the "work in progress" label mean you've got that resolved?

I'm asking because this PR can only be cherry picked to the release branch if it doesn't introduce crashes.

**Comment by dbenage-cx** [comment]
@Vezzra Yes, they were resolved by storing u8 into std::string instead of passing u8.

**Comment by Dilvish-fo** [comment]
It looks like it was a unittest failing here, triggered by an AI client exception, which seems I've seen before recently and certainly is unrelated to this PR:
```
02:35:51.841100 [error] python : configure_logging.py:197 : Uncaught exception: Traceback (most recent call last):
02:35:51.841132 [error] python : configure_logging.py:197 :   File "/freeorion/default/python/AI/FreeOrionAI.py", line 382, in <module>
02:35:51.841144 [error] python : configure_logging.py:197 :     init_handlers(fo.getOptionsDBOptionStr("ai-config"), fo.getAIDir())
02:35:51.841151 [error] python : configure_logging.py:197 :   File "/freeorion/default/python/AI/freeorion_tools/handlers.py", line 10, in init_handlers
02:35:51.841158 [error] python : configure_logging.py:197 :     handlers = split(get_option_dict()[HANDLERS])
02:35:51.841164 [error] python : configure_logging.py:197 : KeyError: 'handlers'
02:35:51.841170 [error] python : configure_logging.py:197 : 
```

So I don't think that should hold up this PR.  But I expect that was a weird flukey failure (which we really should try to figure out better why it's periodically happening), so I'll see if I can finally figure out how to restart the Travis CI tests.

**Comment by Dilvish-fo** [comment]
OK, well at least I finally figured out why I wasn't seeing the restart option before-- I had to log into https://travis-ci.org/profile in order for it to sync up with my github account.

**Pull request merged**